src/feasibility_model.jl

```diff
@@ -0,0 +1,93 @@
+export FeasibilityModel
+
+# TODO: Extend to handle bounds
+"""
+A viability model is created from a NLPModel of the form
+
+    min f(x)
+    s.t c(x) = 0
+
+by defining the function F(x) = c(x). If the problem has
+bounds on the variables or more constraints, an error
+is thrown.
```

Comment:
Normally, what we call a feasibility problem is a problem with a constant objective function. It can have equality and inequality constraints. Its sole purpose is to identify a feasible point. I wonder if we could give this one another name to avoid confusion. Maybe FeasibilityResidual or something like that?